swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/AbstractSwaggerGenerator.java

```diff
@@ -81,9 +84,25 @@
 
   protected String httpMethod;
 
+
+  class SwaggerExt extends Swagger {
+    @Override
+    public void addDefinition(String key, Model model) {
+      if (this.definitions == null) {
+        this.definitions = new LinkedHashMap<>();
+      }
+      if (this.definitions.containsKey(key) && !this.definitions.get(key).equals(model)) {
+        throw new IllegalArgumentException(
+            "the parameter typeName " + key + " duplicate in " + cls.getName()
+                + ", need to rename it");
```

Comment:
throw new IllegalArgumentException("Model name can not be same for different class. Name=“ + key + ", Class=" + cls.getName);